libsodium-bindings/src/LibSodium/Bindings/SecureMemory.hs

```diff
@@ -178,3 +179,11 @@ foreign import capi "sodium.h sodium_free"
     :: forall a
      . Ptr a
     -> IO ()
+
+-- | Function pointer to use as 'Foreign.ForeignPtr' finalizer for sodium-allocated memory.
+--
+-- The memory region is filled with zeros before the deallocation.
+foreign import ccall "sodium.h &sodium_free"
+  finalizerSodiumFree
+    :: forall a
+     . FinalizerPtr a
```

Comment:
Thanks for writing all this for us. As of right now we highly prefer `capi` over `ccall`.

Reply:
Indeed, unless there is a problem with GHC, we try to stick to `capi`: https://github.com/haskell-cryptography/libsodium-bindings/blob/main/libsodium-bindings/CONTRIBUTING.md#c-ffi

Reply:
According to [finalizer docs](https://hackage.haskell.org/package/base-4.18.0.0/docs/GHC-ForeignPtr.html#t:FinalizerPtr) this must be a `ccall`:

> Note that the foreign function must use the ccall calling convention.

Reply:
I investigated a bit and actually capi is fine too: https://gitlab.haskell.org/ghc/ghc/-/issues/23599

The thing is that the capi calling convention generates a C stub that calls to the named function (that we foreign import). The problem is with the `stdcall` and `js` conventions, where these functions get called by the RTS directly.